### ISSUE TITLE
Time / Name components

### ISSUE THREAD

**Issue by delaneyj** [DavidPeicho/ecstra#20]
It appears you are treating entity name and time as a special case.  I've found having a global "singleton" entity with a TimeConponent advantageous as systems can conditional manipulate time given events.  Same with the createEntity(name) is possibly better suited to a `NameComponent{name:string}`.  It makes things like stats generation much easier if there is no special case and the system is generic as possible.

**Comment by DavidPeicho** [comment]
I agree at least for the name. I am thinking of actually just having entity IDs, generated or set by the user. Right now I have a name, as well as an auto-generated ID.

I am not really treating time right now in any special case. I just accept the delta time in the systems. I think you would still be able to use your singleton and your own time component with this
